### PR TITLE
fix(javascript): indentation for sub-ternaries

### LINKED ISSUE #3384: wrong indent (tab width) when using ternary in jsx
**Prettier 1.7.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAxCFMC8mAFAJSEB8JAOlJvZgDwAmAlgG4W0M+bABmeQgSJQArgBsJ3XrID8maiCUzZPJJkH4RRGACcxcVWp4KlATThoVdE7w1KAchCUBfY4wD0bTrVIBuWhAAGhAIAAcYVnRkUABDDGR+OIk0OFCAIz04sABrOBgAZXCc1igAc2R9Q1DmCDAklLTQgCs0AA8AIWy8gsK4gFs4ABkyuEbU9LCxGHCZgCYJ5pASvTS9ZBAMuIyATwloEJW9MpgAdVZmGAALZAAWAAZQtIHWKoMptDLyiTgARTEEHgSymMB2FyutyQd1C+jirAk3wAwhABgM4psoNBxqExGkACo7NAg1yuIA)
```sh
--print-width 40
--tab-width 4
```

**Input:**
```jsx
const Foo = () => (
    <div>
        {foo === null
            ? ""
            : foo === true
              ? "Yes"
              : "No"}
    </div>
);

```

**Output:**
```jsx
const Foo = () => (
    <div>
        {foo === null
            ? ""
            : foo === true
              ? "Yes"
              : "No"}
    </div>
);

```

**Expected behavior:**
```diff
 const Foo = () => (
     <div>
         {foo === null
             ? ""
             : foo === true
-              ? "Yes"
-              : "No"}
+               ? "Yes"
+               : "No"}
     </div>
 );

```

BTW thank you all!

### PR DESCRIPTION
**Pull request by ikatyang**
Fixes #3384 

Only affects `tabWidth > 2`.

---

**Prettier pr-3747**
[Playground link](https://deploy-preview-3747--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuc0DOMAEAxCFMC8mAFAJSEB8JAOlJvZgDwAmAlgG4W0M+bABmeQgSJQArgBsJ3XrID8maiCUzZPJJkH4RRGACcxcVWp4KlATThoVdE7w1KAchCUBfY4wD0bTrVIBuEAAaEAgABxhWdGRQML1WWAB1VmYYAAtkABYABhCYAEMAI2TUjKRMkLQEgHMJOABFMQh4ZH58iTQ4PL181gkagGEIAFth-OQQKGg4YJBCnrAAazgYAGUw-LAa5H1DEIArNAAPACEF5bX84bgAGQSZpDaOrpANvU69CcKigE8JaFmnWGrB2BheYk6ABUimhWu1OiEEh8YAAFHrVMZw54hPRwACOYlYuLR+Qx40e8JecQgnUSPTCEziVjgenYMxC+T0eggAHcUZyELCUPl2BAUrNmBAwFiESB8hgZS8IGIYGEVQAmRWuVxAA)
```sh
--print-width 40
--tab-width 4
```

**Input:**
```jsx
const Foo = () => (
    <div>
        {foo === null
            ? ""
            : foo === true
              ? "Yes"
              : "No"}
    </div>
);
```

**Output:**
```jsx
const Foo = () => (
    <div>
        {foo === null
            ? ""
            : foo === true
                ? "Yes"
                : "No"}
    </div>
);

```

### CODE REVIEW

File: tests/ternaries/indent.js [hunk 1]
```diff
@@ -1 +1,15 @@
 aaaaaaaaaaaaaaa ? bbbbbbbbbbbbbbbbbb : ccccccccccccccc ? ddddddddddddddd : eeeeeeeeeeeeeee ? fffffffffffffff : gggggggggggggggg
+
+aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```
Comment (azz) [inline tests/ternaries/indent.js]: You could set `printWidth: 40` instead of very long strings.

Comment (ikatyang) [reply tests/ternaries/indent.js]: It'll cause unnecessary diffs, I think it'd be better to not touch unrelated snapshots?

Comment (azz) [reply tests/ternaries/indent.js]: Yeah that's true.

File: tests/ternaries/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -255,6 +401,33 @@ a =>
...
+	a
+		? () => {
+				a;
+			}
+		: () => {
+				a;
+			};
+a => (a ? a : a);
+a =>
+	a ? aasdasdasdasdasdasdaaasdasdasdasdasdasdasdasdasdasdasdasdasdaaaaaa : a;
```
Comment (azz) [inline tests/ternaries/__snapshots__/jsfmt.spec.js.snap]: Is there just one space at the start of this line?

Comment (ikatyang) [reply tests/ternaries/__snapshots__/jsfmt.spec.js.snap]: 1 tab, it's the current behavior.

You can see 37b2ece for actual changes related to this PR.

File: tests/ternaries/jsfmt.spec.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 run_spec(__dirname, ["flow", "typescript"]);
 run_spec(__dirname, ["flow", "typescript"], { tabWidth: 4 });
 run_spec(__dirname, ["flow", "typescript"], { useTabs: true });
+run_spec(__dirname, ["flow", "typescript"], { useTabs: true, tabWidth: 4 });
```
Comment (ikatyang) [inline tests/ternaries/jsfmt.spec.js]: I actually want to test `tabWidth: 4`, not sure why I wrote `2` originally...

**Pull request merged**